dandiapi/api/models/version.py

```diff
@@ -221,8 +225,12 @@ def _populate_metadata(self, version_with_assets: Version = None):
         }
         if version_with_assets.id:
             try:
+                assets: models.QuerySet[Asset] = version_with_assets.assets
                 summary = aggregate_assets_summary(
-                    [asset['metadata'] for asset in version_with_assets.assets.values('metadata')]
+                    # There is no limit to how many assets a dandiset can have, so use
+                    # `values_list` and `iterator` here to keep the memory footprint
+                    # of this list low.
+                    assets.values_list('metadata', flat=True).iterator()
```

Comment:
and having Python generator instead of `list`, i.e.

```suggestion
                    (asset['metadata'] for asset in version_with_assets.assets.values('metadata'))
```

would not be *as good*? (although I already like seeing a single `'metadata'` just want to make sure I understand what is happening)

Reply:
@yarikoptic Using `version_with_assets.assets.values('metadata')` caused a `list` of the following format to be loaded all at once into memory:
```
[
	{
		'metadata': { ..... }
	},
	{
		'metadata': { ..... }
	},
	{
		'metadata': { ..... }
	},
 ....
]
```
So, the list comprehension was needed to flatten the list into the correct format for the dandi-schema `aggregate_assets_summary` function. I've changed this instead to use `assets.values_list('metadata', flat=True)`, which accomplishes flattening the list via SQL without needing the additional list comprehension.